samza-api/src/main/java/org/apache/samza/checkpoint/CheckpointManager.java

```diff
@@ -26,6 +26,14 @@
  * implementation-specific location.
  */
 public interface CheckpointManager {
+  /**
+   * Bootstrap any required resources needed prior to start.
+   */
+  default void init() { }
```

Comment:
Why do we need an additional init() here? We already have start().

Reply:
The `init()` method gets called in job coordinators so that the `KafkaCheckpointManager` can create the checkpoint stream there. A better name could be `createStreams()`, but I wasn't sure if that was generic enough for the CheckpointManager interface. Any suggestions on a name or does `createStreams()` work?
The problem with the `start()` method is that it gets called in the `OffsetManager.start()` method which is called in the `SamzaContainer.run()` method. That happens too late for checkpoint creation.